pkg/local-disk-manager/filter/filter_disk.go

```
@@ -19,26 +19,30 @@ const (
 )
 
 type LocalDiskFilter struct {
-	localDisk *v1alpha1.LocalDisk
-	Result    Bool
+	localDisk     *v1alpha1.LocalDisk
+	Result        Bool
+	FailedMessage map[string]struct{}
 }
 
 func NewLocalDiskFilter(ld *v1alpha1.LocalDisk) LocalDiskFilter {
 	return LocalDiskFilter{
-		localDisk: ld,
-		Result:    TRUE,
+		localDisk:     ld,
+		Result:        TRUE,
+		FailedMessage: make(map[string]struct{}),
 	}
 }
 
 func (ld *LocalDiskFilter) Init() *LocalDiskFilter {
 	ld.Result = TRUE
+	ld.FailedMessage = make(map[string]struct{})
 	return ld
 }
 
 func (ld *LocalDiskFilter) Available() *LocalDiskFilter {
 	if ld.localDisk.Status.State == v1alpha1.LocalDiskAvailable {
 		ld.setResult(TRUE)
 	} else {
+		ld.FailedMessage[v1alpha1.LocalDiskAssignFailReasonUnAvailable] = struct{}{}
```

ignore this error may make events more clear, reasons same with `NodeName`